python/ambassador/envoy/v2/v2ready.py

```diff
@@ -0,0 +1,153 @@
+# Copyright 2021 Datawire. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License
+
+from typing import List, TYPE_CHECKING
+
+from .v2listener import V2Listener
+
+if TYPE_CHECKING:
+    from . import V2Config # pragma: no cover
+
+
+class V2Ready(dict):
+
+    @classmethod
+    def generate(cls, config: 'V2Config') -> None:
+        # Inject the ready listener to the list of listeners if enabled
+        rport = config.ir.aconf.module_lookup('ambassador', 'ready_port', -1)
+        if rport <= 0:
+            config.ir.logger.info(f"V2Ready: ==== disabled")
+            return
+
+        rip = config.ir.aconf.module_lookup('ambassador', 'ready_ip', '127.0.0.1')
+        rlog = config.ir.aconf.module_lookup('ambassador', 'ready_log', True)
+
+        config.ir.logger.info(f"V2Ready: ==== listen on %s:%s" % (rip, rport))
```

Comment:
I know it's just a log-line, but this'll be hard to read if `rip` is an IPv6.  Do we have a handy function in Python that mirrors Go's `net.JoinHostPort`?